File: ChatApp.Service/DTOs/User/UserDto.cs
using System;
using System.Collections.Generic;

namespace ChatApp.Service.DTOs
{
    public class UserAddDto
    {
        public string Username { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; } 
        public List<string> Roles { get; set; }
        public UserAddDto()
        {
            this.Roles = new List<string>();
        } 
    }

    public class UserUpdateDto : EntityUpdateDto<Guid>
    {
        public string Username { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; } 
        public string IdentityUserID { get; set; }
        public List<string> Roles { get; set; }
        public UserUpdateDto()
        {
            this.Roles = new List<string>();
        } 
    }

    public class UserCardDto : EntityGetDto<Guid>
    {
        public string Username { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; } 
        public string IdentityUserID { get; set; }
        public List<string> Roles { get; set; }
        public UserCardDto()
        {
            this.Roles = new List<string>();
        }
    }


}

File: ChatApp.Mobile/src/app/DTOs/User/UserDto.ts
import { BaseAddDto, BaseCardDto, BaseUpdateDto } from "../Base/BaseCardDto";

export class UserAddDto extends BaseAddDto {
    username: string;
    name: string;
    surname: string;
    password: string;
    confirmPassword: string;
    phone: string;
    email: string;
    saysisID: string;
    rolesDto: Array<any> = new Array<any>();
    roles: Array<string> = new Array<string>();
}
export class UserCardDto extends BaseCardDto {
    username: string;
    name: string;
    surname: string;
    phone: string;
    email: string;
    saysisID: string;
    identityUserID: string;
    roles: Array<string> = new Array<string>();
}
export class UserUpdateDto extends BaseUpdateDto {
    username: string;
    name: string;
    surname: string;
    phone: string;
    email: string;
    saysisID: string;
    identityUserID: string;
    rolesDto: Array<any> = new Array<any>();
    roles: Array<string> = new Array<string>();
}

